packages/app/src/systems/Asset/components/AssetInput/AmountInput.tsx

```diff
@@ -0,0 +1,101 @@
+import { cssObj } from '@fuel-ui/css';
+import { Button, Flex, Input, Text } from '@fuel-ui/react';
+import { DECIMAL_UNITS } from 'fuels';
+import type { BN } from 'fuels';
+import { useState } from 'react';
+import type { FC, ChangeEvent } from 'react';
+
+import { AmountInputLoader } from './AmountInputLoader';
+
+export type AmountInputProps = {
+  amount: BN;
+};
+
+type AmountInputComponent = FC<AmountInputProps> & {
+  Loader: typeof AmountInputLoader;
+};
+
+export const AmountInput: AmountInputComponent = ({ amount }) => {
```

Comment:
We are missing some props here to customize the component; From top of my head;
1. value (the amount on the input)
2. onChange (when the value is update we need to dispatch a func for the parent component)
3. balance (the balance that should be shown on the input)

You should also add this on the story book to see it working.

In the way it's made right now the amount with be set just once and the parent component will never know the new value.

Reply:
Just for clarification...
`value` is the displayed input in component?  ie what the user inputs into the component
onChange - this will replace the handleAmount change function?
balance - this is what is shown underneath the max button?  We get this value from the amount prop so I'm not sure why we would also pass in the balance

Reply:
I guess amount should probably be renamed to balance

Reply:
Thanks that makes sense I think I understand how to implement these changes